Mathlib/Geometry/Manifold/Algebra/SmoothGerm.lean

```diff
@@ -0,0 +1,149 @@
+/-
+Copyright (c) 2023 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot
+-/
+import Mathlib.Geometry.Manifold.Algebra.SmoothFunctions
+import Mathlib.Order.Filter.Germ
+
+/-!
+# Germs of smooth functions
+
+Germs of smooth functions between smooth manifolds.
+
+## Main definitions and results
+
+* `smoothGerm I I' N x`: the set of germs of smooth functions `f : M → N` at `x : M`
+* `smoothGerm.subring` and friends: if `R` is a smooth ring,
+the space of germs of smooth functions `M → R` is a subring of `Germ (𝓝 x) R`.
+There are analogous versions for additive and multiplicative monoids and groups.
+
+## Tags
+
+germ, smooth function, manifold
+-/
+
+noncomputable section
+
+open Filter Set
+open scoped Manifold Topology
+
+variable {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  -- Declare a smooth manifold `M` over the pair `(E, H)` with model `I`.
+  {E : Type*}
+  [NormedAddCommGroup E] [NormedSpace 𝕜 E] {H : Type*} [TopologicalSpace H]
+  (I : ModelWithCorners 𝕜 E H) {M : Type*} [TopologicalSpace M] [ChartedSpace H M]
+  [SmoothManifoldWithCorners I M]
+  -- Declare a smooth manifold `N` over the pair `(E', H')` with model `I'`.
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace 𝕜 E']
+  {H' : Type*} [TopologicalSpace H'] (I' : ModelWithCorners 𝕜 E' H')
+  {N : Type*} [TopologicalSpace N] [ChartedSpace H' N]
+
+variable (N) in
+/-- The set of all germs of smooth functions `M → N` at `x : N`. -/
+def smoothGerm (x : M) : Set (Germ (𝓝 x) N) :=
+  { Filter.Germ.ofFun f | f : SmoothMap I I' M N }
```

Comment:
Can some differential geometer (@PatrickMassot, @hrmacbeth?) confirm/infirm that we really want smooth germs to be a set of germs rather than a `structure`? I find this design a bit odd given how the rest of the file uses `smoothGerm` (namely as a type), but I am also not knowledgeable enough to make a call either direction.

Reply:
@grunweg During a triage party, this question came up again. We think it should be a type. Can you convince us otherwise?